src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -1422,6 +1422,32 @@ public ReadOnlyCollection<RegisterGroup> GetRegisterGroups()
             await _breakpointManager.EnableAfterFuncEvalAsync();
         }
 
+        /// <summary>
+        /// Finds the line associated with a start address.
+        /// </summary>
+        public async Task<uint> LineForStartAddress(string file, ulong startAddress)
+        {
+            List<ulong> addresses = new List<ulong>();
+            SourceLine[] srcLines = await SourceLineCache.GetLinesForFile(file);
+            if (srcLines == null || srcLines.Length == 0)
+            {
+                srcLines = await SourceLineCache.GetLinesForFile(System.IO.Path.GetFileName(file));
+            }
+            if (srcLines == null || srcLines.Length == 0)
+            {
+                return 0;
+            }
+
+            foreach (var l in srcLines)
```

Comment:
there can be a *LOT* of lines in a source file. a dictionary by line number would be better as you could do lookup in constant time.